packages/flutter/lib/src/widgets/restoration.dart

```diff
@@ -0,0 +1,1031 @@
+  /// descendant widgets via [RestorationScope.of].
+  ///
+  /// If this is null, [RestorationScope.of] invoked by descendants will return
+  /// null which is effectively turning of state restoration for this subtree.
+  final RestorationId restorationId;
+
+  @override
+  State<RestorationScope> createState() => _RestorationScopeState();
+}
+
+class _RestorationScopeState extends State<RestorationScope> with RestorationMixin {
+  @override
+  RestorationId get restorationId => widget.restorationId;
+
+  @override
+  void restoreState(RestorationBucket oldBucket) {
+    // Nothing to do.
+    // The bucket gets injected into the widget tree in the build method.
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return UnmanagedRestorationScope(
+      bucket: bucket, // `bucket` is provided by the RestorationMixin.
+      child: widget.child,
+    );
+  }
+}
+
+/// Inserts a provided [RestorationBucket] into the widget tree and makes it
+/// available to descendants via [RestorationScope.of].
+///
+/// {@macro flutter.widgets.restoration.scope}
+///
+/// If [bucket] is null, no restoration bucket is made available to descendant
+/// widgets ([RestorationScope.of] invoked from a descendant will return null).
+/// This effectively turns off state restoration for the subtree because no
+/// bucket for storing restoration data is made available.
+///
+/// See also:
+///
+///  * [RestorationScope], which inserts a bucket obtained from a surrounding
+///    restoration scope into the widget tree and makes it accessible
+///    for descendants via [RestorationScope.of].
+///  * [RootRestorationScope], which inserts the root bucket provided by
+///    the [RestorationManager] into the widget tree and makes it accessible
+///    for descendants via [RestorationScope.of].
+///  * [RestorationMixin], which may be used in [State] objects to manage the
+///    restoration data of a [StatefulWidget] instead of manually interacting
+///    with [RestorationScope]s and [RestorationBucket]s.
+///  * [RestorationManager], which describes the basic concepts of state
+///    restoration in Flutter.
+class UnmanagedRestorationScope extends InheritedWidget {
+  /// Creates an [UnmanagedRestorationScope].
+  const UnmanagedRestorationScope({
+    Key key,
+    this.bucket,
+    Widget child,
+  }) : super(key: key, child: child);
+
+  /// The [RestorationBucket] that this widget will insert into the widget tree.
+  ///
+  /// Descendant widgets may obtain this bucket via [RestorationScope.of].
+  final RestorationBucket bucket;
+
+  @override
+  bool updateShouldNotify(UnmanagedRestorationScope oldWidget) {
+    return oldWidget.bucket != bucket;
+  }
+}
+
+/// Inserts a child bucket of [RestorationManager.rootBucket] into the widget
+/// tree and makes it available to descendants via [RestorationScope.of].
+///
+/// This widget is usually used close to the root of the widget tree to enable
```

Comment:
nitty suggestion:
```suggestion
/// This widget is usually used near the root of the widget tree to enable
```
(just because "close" can mean "near to" or "to render no longer open", and I kept reading it incorrectly)